src/vs/editor/contrib/codeAction/browser/codeActionModel.ts

```diff
@@ -197,12 +205,82 @@ export class CodeActionModel extends Disposable {
+				const startPosition = trigger.selection.getStartPosition();
+
+				const actions = createCancelablePromise(async token => {
+					if (this._settingEnabledNearbyQuickfixes() && trigger.trigger.type === CodeActionTriggerType.Invoke) {
+						const codeActionSet = await getCodeActions(this._registry, model, trigger.selection, trigger.trigger, Progress.None, token);
+
+						if (token.isCancellationRequested) {
+							return emptyCodeActionSet;
+						}
+
+						// Search for quickfixes in the curret code action set.
+						const foundQuickfix = codeActionSet.validActions?.some(action => action.action.kind ? CodeActionKind.QuickFix.contains(new CodeActionKind(action.action.kind)) : CodeActionKind.None);
+
+						if (!foundQuickfix) {
+							const allMarkers = this._markerService.read({ resource: model.uri });
+
+							// If markers exists, and there are no quickfixes found or length is zero, check for quickfixes on that line.
+							if (allMarkers.length > 0 && (!foundQuickfix || codeActionSet.validActions.length === 0)) {
+								const currPosition: Position = trigger.selection.getPosition();
+								let trackedPosition: Position = currPosition;
+								let distance = Number.MAX_VALUE;
+								let toBeModified = false;
+
+								allMarkers.forEach((marker: IMarker) => {
+									const col = marker.endColumn;
+									const row = marker.endLineNumber;
+									const startRow = marker.startLineNumber;
+
+									// Found quickfix on the same line and check relative distance to other markers
+									if ((row === currPosition.lineNumber || startRow === currPosition.lineNumber) && Math.abs(currPosition.column - col) < distance) {
+										distance = Math.abs(currPosition.column - col);
+										toBeModified = true;
+										trackedPosition = new Position(row, col);
+									}
+								});
+
+								// Only retriggers if actually found quickfix on the same line as cursor
+								if (toBeModified) {
+
+									const newCodeActionTrigger: CodeActionTrigger = {
+										type: trigger.trigger.type,
+										triggerAction: trigger.trigger.triggerAction,
+										filter: { include: CodeActionKind.QuickFix },
+										autoApply: trigger.trigger.autoApply,
+										context: { notAvailableMessage: '', position: trackedPosition }
+									};
+
+									const selectionAsPosition = new Selection(trackedPosition.lineNumber, trackedPosition.column, trackedPosition.lineNumber, trackedPosition.column);
+									const actionsAtMarker = await getCodeActions(this._registry, model, selectionAsPosition, newCodeActionTrigger, Progress.None, token);
+									if (actionsAtMarker.validActions.length !== 0) {
+
+										actionsAtMarker.validActions.forEach(action => {
+											action.toMark = action.action.isPreferred;
+										});
+
+										// Already filtered through to only get quickfixes, so no need to filter again.
+										codeActionSet.validActions.push(...actionsAtMarker.validActions);
```

Comment:
Here it will be safer to use `CodeAction.diagnostics` to filter out any code actions that don't mark themselves as fixing the diagnostic you asked about: https://github.com/microsoft/vscode/blob/2aadaee30802bd528698fe128c358e81396e3c32/src/vscode-dts/vscode.d.ts#L2597